Release 0.6.0 notFailedTemplate throwing a panic when no error

How to reproduce: pretty much any template that throws not errors and contains `notFailedTemplate: {}`


Error 1

![Screenshot 2024-08-31 at 11 25 46](https://github.com/user-attachments/assets/7ec5033c-2b44-4941-97f3-3372be7f7c21)


Error 2

```
panic: interface conversion: interface {} is nil, not string

goroutine 1 [running]:
github.com/helm-unittest/helm-unittest/pkg/unittest/validators.FailedTemplateValidator.validateErrorMessage({{0x0?, 0x38d1400?}, {0x0?, 0x38d1400?}}, {0x0?, 0x0?}, 0x0?, 0xc00064bf80?, 0x10?, {0x46b3dc0, ...})
	/home/runner/work/helm-unittest/helm-unittest/pkg/unittest/validators/failed_template_validator.go:122 +0x21c
github.com/helm-unittest/helm-unittest/pkg/unittest/validators.FailedTemplateValidator.validateManifests({{0x0?, 0x247865e?}, {0x0?, 0xc0004b2390?}}, {0xc0003f6298, 0x1, 0xc000371188?}, 0xc00047b310)
	/home/runner/work/helm-unittest/helm-unittest/pkg/unittest/validators/failed_template_validator.go:88 +0x274
github.com/helm-unittest/helm-unittest/pkg/unittest/validators.FailedTemplateValidator.Validate({{0x0?, 0x241e3a5?}, {0x0?, 0x39975a0?}}, 0x98d7387165255101?)
	/home/runner/work/helm-unittest/helm-unittest/pkg/unittest/validators/failed_template_validator.go:65 +0x3ac
github.com/helm-unittest/helm-unittest/pkg/unittest.(*Assertion).validateTemplate(0xc0004915e0, {0xc0003f6298, 0x1, 0x1}, {0xc0003f6298, 0x1, 0x1}, {0x3a90468, 0xc0006ef900}, {0x0, ...})
	/home/runner/work/helm-unittest/helm-unittest/pkg/unittest/assertion.go:125 +0x145
github.com/helm-unittest/helm-unittest/pkg/unittest.(*Assertion).Assert(0xc0004915e0, 0xc0004b2360, {0x3a90468, 0xc0006ef900}, 0x1, {0x0, 0x0}, 0xc00047b2c0)
	/home/runner/work/helm-unittest/helm-unittest/pkg/unittest/assertion.go:99 +0x894
github.com/helm-unittest/helm-unittest/pkg/unittest.(*TestJob).runAssertions(0xc000494900?, 0xc0004b2360, {0x3a90468, 0xc0006ef900}, 0x1, {0x0, 0x0}, 0x0)
	/home/runner/work/helm-unittest/helm-unittest/pkg/unittest/test_job.go:445 +0x157
github.com/helm-unittest/helm-unittest/pkg/unittest.(*TestJob).RunV3(0xc000494900, 0xc00033aaa0, 0xc000335940, 0x0, {0x0, 0x0}, 0xc000183630)
	/home/runner/work/helm-unittest/helm-unittest/pkg/unittest/test_job.go:222 +0x337
github.com/helm-unittest/helm-unittest/pkg/unittest.(*TestSuite).runV3TestJobs(0xc0001e6280, {0x7ff7beaf0b63, 0x17}, 0xc000335940, 0x0, {0x0, 0x0})
	/home/runner/work/helm-unittest/helm-unittest/pkg/unittest/test_suite.go:346 +0x165
github.com/helm-unittest/helm-unittest/pkg/unittest.(*TestSuite).RunV3(0xc0001e6280, {0x7ff7beaf0b63, 0x17}, 0xc000335940, 0x0, {0x0, 0x0}, 0xc0002fd5e0)
	/home/runner/work/helm-unittest/helm-unittest/pkg/unittest/test_suite.go:230 +0xca
github.com/helm-unittest/helm-unittest/pkg/unittest.(*TestRunner).runV3SuitesOfChart(0x46884c0, {0xc000470000, 0x2a, 0x7ff7beaf0b63?}, {0x7ff7beaf0b63, 0x17})
	/home/runner/work/helm-unittest/helm-unittest/pkg/unittest/test_runner.go:223 +0x22c
github.com/helm-unittest/helm-unittest/pkg/unittest.(*TestRunner).RunV3(0x46884c0, {0xc00031fc80, 0x1, 0x0?})
	/home/runner/work/helm-unittest/helm-unittest/pkg/unittest/test_runner.go:130 +0x17c
main.RunPlugin(0xc000146900?, {0xc00031fc80, 0x1, 0x3})
	/home/runner/work/helm-unittest/helm-unittest/cmd/helm-unittest/helm_unittest.go:112 +0x310
github.com/spf13/cobra.(*Command).execute(0x4672000, {0xc000152050, 0x3, 0x3})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:989 +0xa91
github.com/spf13/cobra.(*Command).ExecuteC(0x4672000)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041
main.main()
	/home/runner/work/helm-unittest/helm-unittest/cmd/helm-unittest/helm_unittest.go:121 +0x1a
```

working on fix here https://github.com/helm-unittest/helm-unittest/pull/399

--- a/pkg/unittest/validators/failed_template_validator.go
+++ b/pkg/unittest/validators/failed_template_validator.go
@@ -55,9 +55,11 @@ func (a FailedTemplateValidator) Validate(context *ValidateContext) (bool, []str
 		errorMessage := splitInfof(errorFormat, -1, "single attribute 'errorMessage' or 'errorPattern' supported at the same time")
 		validateErrors = append(validateErrors, errorMessage...)
 	} else if context.RenderError != nil {
-		if a.ErrorMessage != "" && reflect.DeepEqual(a.ErrorMessage, context.RenderError.Error()) == context.Negative && a != (FailedTemplateValidator{}) {
-			errorMessage := a.failInfo(context.RenderError.Error(), -1, context.Negative)
-			validateErrors = append(validateErrors, errorMessage...)
+		// Validating error, when the errorSource is due to rendering errors
+		if a.ErrorPattern != "" {
+			validateSuccess, validateErrors = a.validateErrorPattern(context.RenderError.Error(), -1, context, validateSuccess, validateErrors)
+		} else if a.ErrorMessage != "" {
+			validateSuccess, validateErrors = a.validateErrorMessage(context.RenderError.Error(), -1, context, validateSuccess, validateErrors)
 		} else {
 			validateSuccess = true
 		}
@@ -84,8 +86,10 @@ func (a FailedTemplateValidator) validateManifests(manifests []common.K8sManifes
 
 		if a.ErrorPattern != "" {
 			currentSuccess, validateErrors = a.validateErrorPattern(actual, idx, context, currentSuccess, validateErrors)
-		} else {
+		} else if a.ErrorMessage != "" {
 			currentSuccess, validateErrors = a.validateErrorMessage(actual, idx, context, currentSuccess, validateErrors)
+		} else {
+			currentSuccess = true
 		}
 
 		validateSuccess = determineSuccess(idx, validateSuccess, currentSuccess)
@@ -108,7 +112,7 @@ func (a FailedTemplateValidator) validateErrorPattern(actual interface{}, idx in
 		return validateSuccess, validateErrors
 	}
 
-	if p.MatchString(actual.(string)) == context.Negative {
+	if (actual != nil && p.MatchString(actual.(string))) == context.Negative {
 		errorMessage := a.failInfo(actual, idx, context.Negative)
 		validateErrors = append(validateErrors, errorMessage...)
 	} else {
@@ -119,7 +123,7 @@ func (a FailedTemplateValidator) validateErrorPattern(actual interface{}, idx in
 }
 
 func (a FailedTemplateValidator) validateErrorMessage(actual interface{}, idx int, context *ValidateContext, validateSuccess bool, validateErrors []string) (bool, []string) {
-	if reflect.DeepEqual(a.ErrorMessage, actual.(string)) == context.Negative {
+	if (actual != nil && reflect.DeepEqual(a.ErrorMessage, actual.(string))) == context.Negative {
 		errorMessage := a.failInfo(actual, idx, context.Negative)
 		validateErrors = append(validateErrors, errorMessage...)
 	} else {

--- a/pkg/unittest/validators/failed_template_validator_test.go
+++ b/pkg/unittest/validators/failed_template_validator_test.go
@@ -226,12 +226,17 @@ func TestFailedTemplateValidatorWhenNegativeAndFail(t *testing.T) {
 }
 
 func TestFailedTemplateValidatorWhenRenderError(t *testing.T) {
+	testRenderError := errors.New("values don't meet the specifications of the schema(s)")
 	tests := []struct {
 		name      string
 		validator FailedTemplateValidator
 	}{
 		{
-			name:      "Test case 1: with error message",
+			name:      "Test case 1: with error pattern",
+			validator: FailedTemplateValidator{ErrorPattern: "schema"},
+		},
+		{
+			name:      "Test case 2: with error message",
 			validator: FailedTemplateValidator{ErrorMessage: "values don't meet the specifications of the schema(s)"},
 		},
 		{
@@ -245,7 +250,7 @@ func TestFailedTemplateValidatorWhenRenderError(t *testing.T) {
 
 			pass, diff := tt.validator.Validate(&ValidateContext{
 				Docs:        []common.K8sManifest{},
-				RenderError: errors.New(tt.validator.ErrorMessage),
+				RenderError: testRenderError,
 			})
 
 			assert.True(t, pass)
@@ -254,7 +259,7 @@ func TestFailedTemplateValidatorWhenRenderError(t *testing.T) {
 	}
 }
 
-func TestFailedTemplateValidatorWhenErrorAndContainsSet(t *testing.T) {
+func TestFailedTemplateValidatorWhenErrorMessageAndErrorPatternSet(t *testing.T) {
 	manifest := makeManifest(failedTemplate)
 	v := FailedTemplateValidator{ErrorMessage: "A field should be required", ErrorPattern: "pattern is set"}
 	pass, diff := v.Validate(&ValidateContext{

--- a/test/data/v3/basic/tests/ingress_test.yaml
+++ b/test/data/v3/basic/tests/ingress_test.yaml
@@ -33,6 +33,7 @@ tests:
       - containsDocument:
           kind: Ingress
           apiVersion: extensions/v1beta1
+      - notFailedTemplate: {}
 
   - it: should set annotations if given
     set:

